src/util/json.h

```diff
@@ -11,5 +12,26 @@
 // most notably to std::string.
 #define JSON_USE_IMPLICIT_CONVERSIONS 0
 #include <nlohmann/json.hpp>
+#include <optional>
+
+// Adding third party type support.
+namespace nlohmann {
+    template <typename T>
+    struct adl_serializer<std::optional<T>> {
+        static void to_json(nlohmann::json& j, const std::optional<T>& opt) {
+          j = nlohmann::json{{"has_value", opt.has_value()}};
+          // Only add, if there is value.
+          if (opt.has_value()) {j["value"] = opt.value();}
+        }
+
+        static void from_json(const nlohmann::json& j, std::optional<T>& opt) {
+          if (j.is_null() || !j["has_value"].get<bool>()){
+            opt = std::nullopt;
+          } else {
+            opt = j["value"].get<T>();
+          }
```

Comment:
There is a null-json.
So I would prefer a much simpler format:
````
if (opt.has_value()) {
 j = opt.value();
} else {
 j = nullptr;
}
```
This is the way we also serialize `std::optional` for query results with undefined values.

Reply:
Done, see [c3690bf](https://github.com/ad-freiburg/qlever/pull/860/commits/c3690bf4635ea3efddbbd00478d6d33e9ccf9903).

Question: If you already serialize `std::optional` for query results, then shouldn't I better use that implementation, instead of my own?